### PR TITLE
[iOS] Model installation - use of package version

### LINKED ISSUE #2192: [iOS] Lexical models fail to install because lexical model version no longer generated
This parallels #2190, but for iOS instead of Android.

**Describe the bug**
User reported issue from [community site](https://community.software.sil.org/t/installation-of-predictive-text-package-in-keyman-for-android/2649)

Lexical models built with Keyman Developer 12.0.50 stable no longer include lexical model versions (they use the lexical model package version).

However, the Android and iOS apps were developed to expect a lexical model version, so this is causing the new model packages to fail at installation.

(The published models on downloads.keyman.com/models haven't been updated with the new lexical model compiler so they currently contain model version info)

Install a Mozhi Malayalam keyboard for testing.

Testing with a model package built from
https://my.pcloud.com/publink/show?code=kZlX3LkZpVxg32Gpk4BRIGIW5M7hhBSpWXik

### Fix
The fix is to use the package version "info/version/description" for the lexical model version.
If package version doesn't exist, fall back to "1.0"

### Configuration
Keyman Developer 12.0.50 Stable
Keyman iOS 12.0.50 Stable

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #2192.

### CODE REVIEW

**Comment by mcdurdin** [comment]
Sorry, can we please split the iOS and Android issues because I would like to track the fix separately.

File: ios/engine/KMEI/KeymanEngine/Classes/Manager.swift [hunk 1]
```diff
@@ -697,11 +697,20 @@ public class Manager: NSObject, UIGestureRecognizerDelegate {
       let data = try Data(contentsOf: path, options: .mappedIfSafe)
       let jsonResult = try JSONSerialization.jsonObject(with: data, options: .mutableLeaves)
       if let jsonResult = jsonResult as? [String:AnyObject] {
+        var version: String = "1.0"
+
+        if let info = jsonResult["info"] as? [String:AnyObject] {
+          if let versionEntry = info["version"] as? [String:AnyObject] {
+            if let description = versionEntry["description"] as? String {
+              version = description;
```
Comment (mcdurdin) [inline ios/engine/KMEI/KeymanEngine/Classes/Manager.swift]: Technically, we need to validate that the string is a dotted numeric (e.g. `1.0.1`) and fall back to `1.0` if not. But willing to let this go for now.

**Comment by mcdurdin** [comment]
in what circs will that guard throw? if version is 'foo'?

**Comment by jahorton** [comment]
Guard will throw when the initializer returns `nil`.  The relevant code:

```
    let stringComponents = string.components(separatedBy: ".")
    var components: [Int] = []
    for s in stringComponents {
      guard let i = Int(s), i >= 0 else {
        return nil
      }
      components.append(i)
    }
```

So, if any subcomponent isn't castable to a non-negative Integer, the initializer will fail.

**Comment by mcdurdin** [comment]
So does this mean a package with version “foo” will not install? It should
install but the version should be treated as “1.0”

**Pull request merged**